mkdocs_git_authors_plugin/plugin.py

```diff
@@ -1,46 +1,36 @@
-from mkdocs_git_authors_plugin.git.command import GitCommandError
-import re
 import logging
-from mkdocs.config import config_options
+import re
+from typing import Literal, Union
+
+from mkdocs.config.defaults import MkDocsConfig
 from mkdocs.plugins import BasePlugin
+from mkdocs.structure.files import Files
+from mkdocs.structure.nav import Navigation
+from mkdocs.structure.pages import Page
+from mkdocs.utils.templates import TemplateContext
 
-from . import util
-from .git.repo import Repo
+from mkdocs_git_authors_plugin import util
 from mkdocs_git_authors_plugin.ci import raise_ci_warnings
+from mkdocs_git_authors_plugin.config import GitAuthorsPluginConfig
 from mkdocs_git_authors_plugin.exclude import exclude
+from mkdocs_git_authors_plugin.git.command import GitCommandError
+from mkdocs_git_authors_plugin.git.repo import Repo
 
 logger = logging.getLogger("mkdocs.plugins")
 
 
-class GitAuthorsPlugin(BasePlugin):
-    config_scheme = (
-        ("show_contribution", config_options.Type(bool, default=False)),
-        ("show_line_count", config_options.Type(bool, default=False)),
-        ("show_email_address", config_options.Type(bool, default=True)),
-        ("href", config_options.Type(str, default='mailto:{email}')),
-        ("count_empty_lines", config_options.Type(bool, default=True)),
-        ("fallback_to_empty", config_options.Type(bool, default=False)),
-        ("exclude", config_options.Type(list, default=[])),
-        ("ignore_commits", config_options.Type(str, default=None)),
-        ("ignore_authors", config_options.Type(list, default=[])),
-        ("enabled", config_options.Type(bool, default=True)),
-        ("enabled_on_serve", config_options.Type(bool, default=True)),
-        ("sort_authors_by", config_options.Type(str, default="name")),
-        ("authorship_threshold_percent", config_options.Type(int, default=0)),
-        ("strict", config_options.Type(bool, default=True)),
-        # ('sort_authors_by_name', config_options.Type(bool, default=True)),
-        # ('sort_reverse', config_options.Type(bool, default=False))
-    )
-
-    def __init__(self):
+class GitAuthorsPlugin(BasePlugin[GitAuthorsPluginConfig]):
+    def __init__(self) -> None:
         self._repo = None
         self._fallback = False
         self.is_serve = False
 
-    def on_startup(self, command, dirty):
+    def on_startup(
+        self, command: Literal["build", "gh-deploy", "serve"], dirty: bool
+    ) -> None:
         self.is_serve = command == "serve"
 
-    def on_config(self, config, **kwargs):
+    def on_config(self, config: MkDocsConfig) -> Union[MkDocsConfig, None]:
```

Comment:
Please leave the `**kwargs` arguments in the plugins events. This is recommended by mkdocs for future compatibility: https://www.mkdocs.org/dev-guide/plugins/#config:~:text=It%20is%20good%20practice%20to%20accept%20keyword%20arguments%20as%20**kwargs.%20In%20the%20event%20that%20additional%20keywords%20are%20provided%20to%20an%20event%20in%20a%20future%20version%20of%20MkDocs%2C%20there%20will%20be%20no%20need%20to%20alter%20your%20plugin.